pkg/build/builder/daemonless.go

```diff
@@ -311,118 +328,233 @@ func buildDaemonlessImage(sc types.SystemContext, store storage.Store, isolation
 		PullPushRetryDelay:      DefaultPushOrPullRetryDelay,
 	}
 
-	_, _, err := imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
+	_, _, err = imagebuildah.BuildDockerfiles(opts.Context, store, options, opts.Dockerfile)
 	return err
 }
 
-func generateTransientMounts() []string {
-	mounts := []string{}
-	mounts = appendRHSMMount(defaultMountStart, mounts)
-	mounts = appendETCPKIMount(defaultMountStart, mounts)
-	mounts = appendRHRepoMount(defaultMountStart, mounts)
-	mounts = appendCATrustMount(mounts)
+type TransientMount struct {
+	buildapiv1.BuildVolumeMount
+	MountPathSource string
+	// additional buildah mount options not exposed to the Build API
+	NoDev  bool
+	NoExec bool
+	NoSuid bool
+}
+
+func (tm TransientMount) String() string {
+	var mountOptions []string
+	if !tm.BuildVolumeMount.ReadOnly {
+		mountOptions = append(mountOptions, "rw")
+	}
```

Comment:
Need an else for read-only
```suggestion
	if !tm.BuildVolumeMount.ReadOnly {
		mountOptions = append(mountOptions, "rw")
	} else {
		mountOptions = append(mountOptions, "ro") 
	}
```